ecommerce/views.py

```diff
@@ -124,6 +126,14 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             order.status = Order.FULFILLED
 
         order.save_and_log(acting_user=None)
+
+        try:
+            application = order.applications.get()
+            application.state = AppStates.COMPLETE.value
+            application.save()
```

Comment:
Can you define a `@transition` method on the `BootcampApplication` model and invoke that here instead? https://github.com/viewflow/django-fsm#usage